core/trino-main/src/main/java/io/trino/execution/QueryStateMachine.java

```diff
@@ -647,8 +645,11 @@ QueryInfo getQueryInfo(Optional<StageInfo> rootStage)
                 version);
     }
 
-    private QueryStats getQueryStats(Optional<StageInfo> rootStage, List<StageInfo> allStages)
+    private QueryStats getQueryStats(Optional<StagesInfo> stages)
     {
+        Optional<StageInfo> rootStage = stages.map(StagesInfo::getOutputStage);
```

Comment:
why do we need this variable? please inline :)

Reply:
But this is used many times